pandas/_config/config.py

```diff
@@ -751,7 +756,7 @@ def inner(key, *args, **kwds):
 # arg in register_option
 
 
-def is_type_factory(_type):
+def is_type_factory(_type) -> Callable[..., None]:
```

Comment:
the returned function accepts exactly one argument

```suggestion
def is_type_factory(_type: Type[Any]) -> Callable[[Any], None]:
```